### PR TITLE
Merge 3.2

### PR DESCRIPTION
**Pull request by wallyworld**
Merge 3.2 - a lot of conflicts and bespoke changes needed to due to 3.2->3.3 divergence around code with series.

https://github.com/juju/juju/pull/16000 [from wallyworld/fix-via-check](https://github.com/juju/juju/commit/151bd647502fc8a879e7b0187cec60b349b84b15)
https://github.com/juju/juju/pull/16004 [from SimonRichardson/sort-controller-names-…](https://github.com/juju/juju/commit/cd70af379a023de93598a083d523e250d915614e)
https://github.com/juju/juju/pull/16021 [from barrettj12/rm-kinetic](https://github.com/juju/juju/commit/58e23e32d77825cb7bc6d6e6fe8f458dbdfb9225)
https://github.com/juju/juju/pull/16020 [from barrettj12/cmr-wait](https://github.com/juju/juju/commit/8a5692f008fabf7f06a53f2ee84d8bad06e78501)
https://github.com/juju/juju/pull/16022 [from jack-w-shaw/remove_core_series_to_core…](https://github.com/juju/juju/commit/610187527d715420282445bb7f242d47472fc3e4)
https://github.com/juju/juju/pull/16030 [from jack-w-shaw/JUJU-4367_fix_incorrect_na…](https://github.com/juju/juju/commit/ad7c2e85f0a3072200bbce90f59a20bfa618bf5d)
https://github.com/juju/juju/pull/16035 [from wallyworld/merge-3.1-20230802](https://github.com/juju/juju/commit/bbad9d12de0ebefbd33af942599cc7760c59a1a3)
https://github.com/juju/juju/pull/16034 [from wallyworld/delay-secret-removal](https://github.com/juju/juju/commit/910963ddc56ccbd50414575c6c6ffee9c1f2b9be)

```
# Conflicts:
#       api/client/application/client_test.go
#       apiserver/facades/client/application/deployrepository.go
#       caas/kubernetes/provider/bootstrap.go
#       cmd/juju/application/deployer/bundle.go
#       cmd/juju/application/deployer/bundlehandler.go
#       cmd/juju/application/deployer/charm.go
#       cmd/juju/application/refresher/refresher.go
#       core/bundle/changes/changes.go
#       core/bundle/changes/handlers.go
#       core/bundle/changes/model.go
#       core/charm/repository/charmhub.go
#       go.mod
#       go.sum
```

### CODE REVIEW

File: cmd/juju/application/deploy_test.go [hunk 1]
```diff
@@ -1503,7 +1503,7 @@ func (s *DeploySuite) setupNonESMBase(c *gc.C) (series.Base, string) {
 	}
 	defer func() { _ = file.Close() }()
 
-	// Overwrite the metadata.yaml to contain a non EMS series.
+	// Overwrite the metadata.yaml to contain a non EMS corebase.
```
Comment (jack-w-shaw) [inline cmd/juju/application/deploy_test.go:1506]: Oops

**Review by jack-w-shaw — APPROVED** [review]
LGTM other than the one small mistake

File: core/charm/computedbase.go [hunk 2]
```diff
@@ -8,19 +8,19 @@ import (
 	"github.com/juju/collections/transform"
 	"github.com/juju/errors"
 
-	"github.com/juju/juju/core/series"
+	"github.com/juju/juju/core/base"
 )
 
 // BaseForCharm takes a requested series and a list of series supported by a
 // charm and returns the series which is relevant.
```
[the remark this answers is not in the document]
Comment (wallyworld) [reply core/charm/computedbase.go]: good catch, fixed

File: cmd/juju/application/deploy_test.go (hunk 1 above)
Comment (wallyworld) [reply cmd/juju/application/deploy_test.go]: damn, caught one of these, missed this one

**Pull request merged**